src/main/java/org/reflections8/util/ConfigurationBuilder.java

```diff
@@ -343,10 +343,10 @@ public ConfigurationBuilder addClassLoaders(ClassLoader... classLoaders) {
             ClassLoader[] existing = this.classLoaders.get();
             int existingLength = existing.length;
             ClassLoader[] tmpArray = Arrays.copyOf(existing, existingLength + classLoaders.length);
-            for (int i = 0; i < classLoaders.length; i++) {
-                tmpArray[i + existingLength] = classLoaders[i];
-            }
+            System.arraycopy(classLoaders, 0, tmpArray, existingLength, classLoaders.length);
             this.classLoaders = Optional.of(tmpArray);
+        } else {
+            this.classLoaders = Optional.of(classLoaders);
         }
         return this;
```

Comment:
looks good, 
could you remove "return this;" since all branches of the if are handled

Reply:
Could you add a test for this fix?